finetune/input_pipeline.py

```diff
@@ -371,19 +387,40 @@ def get_train_input_fns(self, Xs, Y=None, batch_size=None, val_size=None, contex
             )
         else:
             self._skip_tqdm = 0
-            if context is not None:
-                to_shuffle = (Xs, Y, context)
+            if context is not None and log_proba_biases is not None:
```

Comment:
Can we pack everything together into a list of dict or placeholder the tuples with lists of Nones and split on that to reduce the number of code paths?

Reply:
Sounds good :+1: